win32/WatchData.h

```diff
@@ -41,38 +39,35 @@
 
 using namespace std;
 
-typedef void(*ChangeCallback)(int watchID, int action, const WCHAR* rootPath, const WCHAR* filePath);
+typedef void(*ChangeCallback)(int watchID, int action, const WCHAR* filePath, int len);
 
 class WatchData
 {
-private:	
-	static int _counter;
-	WCHAR* _path;
-	int _watchId;
-	HANDLE _hDir;
-	int _mask;
-	bool _watchSubtree;
-	DWORD _byteReturned;
-	OVERLAPPED _overLapped;
-	char _buffer[8196];
-	HANDLE _completionPort;
+private:
+    static int _counter;
+
+    bool _watchSubtree;
```

Comment:
I recommend locating this later, because that bool is going to take up 4 bytes because C struct alignment